src/main/java/com/openlattice/search/SearchService.java

```diff
@@ -224,24 +205,37 @@ public void createEntitySet( EntitySetCreatedEvent event ) {
     @Timed
     @Subscribe
     public void deleteEntitySet( EntitySetDeletedEvent event ) {
-        elasticsearchApi.deleteEntitySet( event.getEntitySetId() );
+        elasticsearchApi.deleteEntitySet( event.getEntitySetId(), event.getEntityTypeId() );
     }
 
     @Timed
     @Subscribe
     public void deleteEntities( EntitiesDeletedEvent event ) {
-        event.getEntitySetIds().forEach(
-                entitySetId -> event.getEntityKeyIds()
-                        .stream()
-                        .map( id -> new EntityDataKey( entitySetId, id ) )
-                        .forEach( elasticsearchApi::deleteEntityData )
-        );
+        UUID entityTypeId = dataModelService.getEntityTypeByEntitySetId( event.getEntitySetId() ).getId();
+        elasticsearchApi.deleteEntityDataBulk( event.getEntitySetId(), entityTypeId, event.getEntityKeyIds() );
+    }
+
+    @Timed
+    @Subscribe
+    public void deleteLinkedEntities( LinkedEntitiesDeletedEvent event ) {
+        if ( event.getEntitySetIds().size() > 0 ) {
+            UUID entityTypeId = dataModelService.getEntityTypeByEntitySetId( event.getEntitySetIds().iterator().next() )
```

Comment:
Entity types can differ here by linking entity set (event hold linking entity sets, not linked ones)

Reply:
true, updated -- thanks!